### PR TITLE
CSharp "Hello World" application

### PR DESCRIPTION
**Pull request by AFFogarty**
### What changes were proposed in this pull request?
This change is a recreation of #90 in CSharp. It addresses issue #79 .

### Why are the changes needed?
To provide a sample application in .NET.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
This change consists of an standalone example application and is verified by running it manually and validating the output and its behavior.

### CODE REVIEW

File: examples/csharp/HyperspaceApp/Program.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            // Create Spark session.
+            SparkSession spark = SparkSession
+                .Builder()
+                .AppName("Hyperspace example")
+                .Config("spark.some.config.option", "some-value")
```
Comment (apoorvedave1) [inline examples/csharp/HyperspaceApp/Program.cs:33]: remove this?

File: examples/csharp/HyperspaceApp/.gitignore [hunk 2]
```diff
@@ -0,0 +1,369 @@
+## Ignore Visual Studio temporary files, build results, and
```
Comment (apoorvedave1) [inline examples/csharp/HyperspaceApp/.gitignore:1]: can we merge this .gitignore file to <root>/.gitignore ? Unless it's required here?

File: examples/csharp/HyperspaceApp/Program.cs (hunk 1 above)
Comment (AFFogarty) [reply examples/csharp/HyperspaceApp/Program.cs:33]: This is copied from the Scala example:  https://github.com/microsoft/hyperspace/blob/master/examples/scala/src/main/scala/App.scala#L29

File: examples/csharp/HyperspaceApp/.gitignore (hunk 2 above)
Comment (AFFogarty) [reply examples/csharp/HyperspaceApp/.gitignore:1]: Hmm... the rules in this file are all specific to Visual Studio / .NET so I'm wondering if we should keep them separate?

Comment (imback82) [reply examples/csharp/HyperspaceApp/.gitignore:1]: I am fine leaving these separate (maybe under csharp)?

File: examples/csharp/HyperspaceApp/Program.cs (hunk 1 above)
Comment (pirz) [reply examples/csharp/HyperspaceApp/Program.cs:33]: @apoorvedave1 This follows what Spark examples do. Check [here](https://github.com/apache/spark/blob/master/examples/src/main/scala/org/apache/spark/examples/sql/SparkSQLExample.scala#L39) as an example.

**Comment by pirz** [comment]
@AFFogarty LGTM.

File: examples/csharp/HyperspaceApp/.gitignore (hunk 2 above)
Comment (AFFogarty) [reply examples/csharp/HyperspaceApp/.gitignore]: Moved to `csharp` folder.

**Comment by imback82** [comment]
Should we update the azure pipeline to make sure this compiles? :)

File: examples/csharp/HyperspaceApp/Program.cs (hunk 1 above)
Comment (apoorvedave1) [reply examples/csharp/HyperspaceApp/Program.cs:33]: oh interesting. Thanks!

**Comment by apoorvedave1** [comment]
LGTM, Thanks @AFFogarty

**Comment by imback82** [comment]
Looks like the Scala example #90 also missed setting up the build. I will create an issue and we can do a follow up PR.

**Pull request merged**